trie/database.go

```diff
@@ -792,13 +792,12 @@ func (db *Database) Update(nodes *MergedNodeSet) error {
 	}
 	for _, owner := range order {
 		subset := nodes.sets[owner]
-		for _, path := range subset.updates.order {
-			n, ok := subset.updates.nodes[path]
-			if !ok {
-				return fmt.Errorf("missing node %x %v", owner, path)
+		subset.forEachWithOrder(func(path string, n *memoryNode) {
+			if n.isDeleted() {
+				return // ignore deletion
```

Comment:
Hmm, why do we ignore this?

Reply:
I guess the old code kind of also didn't have deletions? Still, what happened previously with deletions?

Reply:
In hash-based scheme, we never do deletion. The deleted nodes will be kept in database as dangling ones.
A strong reason behind it is the node might also be referenced by other trie(e.g. storage) and we have no
clue that if it's deletable.